flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/agg/LeadLagAggITCase.scala

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.runtime.batch.sql.agg
+
+import org.apache.flink.table.api.config.ExecutionConfigOptions
+import org.apache.flink.table.api.{DataTypes, Table}
+import org.apache.flink.table.planner.runtime.utils.BatchTestBase.row
+import org.junit.Test
+
+import scala.collection.Seq
+
+class LeadLagAggITCase extends AggregateITCaseBase("LeadLagAggregate") {
+
+  var testCardinality: Table = _
+  var testAllDataTypeCardinality: Table = _
+
+  override def before(): Unit = {
+    super.before()
+
+    testCardinality = tEnv.fromValues(
```

Comment:
As we have quite a full set of test data below this small data set is not needed.